### PR TITLE
feat: add Claude 3.5 Sonnet model

### PR DESCRIPTION
**Pull request by cabljac**
Fixes https://github.com/firebase/genkit/issues/437

This PR adds Claude 3.5 Sonnet to the Vertex AI plugin. (Work in progress)

Checklist (if applicable):
- [ ] Tested (manually, unit tested, etc.)
- [ ] Changelog updated
- [ ] Docs updated

### CODE REVIEW

File: js/plugins/vertexai/src/anthropic.ts [hunk 1]
```diff
@@ -87,6 +103,7 @@ export const SUPPORTED_ANTHROPIC_MODELS: Record<
   string,
   ModelReference<typeof GenerationCommonConfigSchema>
 > = {
+  'claude-3.5-sonnet': claude35Sonnet,
```
Comment (pavelgj) [inline js/plugins/vertexai/src/anthropic.ts:106]:
```suggestion
  'claude-3-5-sonnet': claude35Sonnet,
```

**Comment by pavelgj** [comment]
I verified that it works (with the suggested fix). I believe it's only available in limited regions. `us-east5` works for me, but `europe-west1` should also work.

```
    vertexAI({
      location: 'us-east5',
      modelGardenModels: [claude3Sonnet, claude35Sonnet],
    }),
```

**Review by pavelgj — APPROVED** [review]

**Comment by cabljac** [comment]
@pavelgj I didn't realise you had reviewed, and i force pushed some changes. I bumped the Anthropic SDKs and cast things since we don't support anthropic tools yet, is this OK, or should i keep the old version

**Review by pavelgj — APPROVED** [review]

**Comment by pavelgj** [comment]
> @pavelgj I didn't realise you had reviewed, and i force pushed some changes. I bumped the Anthropic SDKs and cast things since we don't support anthropic tools yet, is this OK? or should i revert to the old version of the SDK if it works.
> 
> #293 should be ready soon, which will resolve this

Looks good! Feel free to merge.

**Pull request merged**